expression/builtin_http_client.go

```diff
@@ -0,0 +1,54 @@
+package expression
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/pingcap/tidb/config"
+)
+
+type tidbServerClient struct {
```

Comment:
I think write a new exported class in server package and wrap `tikvHandlerTool` is better.

Reply:
I agree, and I split to a client package with minimal dependencies inside server package, and it could add backoff, client tracing supported to this package in the future.

Detail in [Code](https://github.com/pingcap/tidb/pull/17397/files#diff-f06427cddd35a8a3b14b70de4cc6027c), PTAL, :)